hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ScmClient.java

```diff
@@ -113,12 +118,27 @@ public Map<Long, Pipeline> getContainerLocations(Iterable<Long> containerIds,
     }
     try {
       Map<Long, Pipeline> result = containerLocationCache.getAll(containerIds);
-      // Don't keep empty pipelines in the cache.
-      List<Long> emptyPipelines = result.entrySet().stream()
-          .filter(e -> e.getValue().isEmpty())
+      // Don't keep empty pipelines or insufficient EC pipelines in the cache.
+      List<Long> uncachePipelines = result.entrySet().stream()
+          .filter(e -> {
+            Pipeline pipeline = e.getValue();
+            // filter empty pipelines
+            if (pipeline.isEmpty()) {
+              return true;
+            }
+            // filter insufficient EC pipelines
+            ReplicationConfig repConfig = pipeline.getReplicationConfig();
+            if (repConfig instanceof ECReplicationConfig) {
+              int d = ((ECReplicationConfig) repConfig).getData();
+              Set<Integer> dataIndexes = EC_DATA_INDEXES.computeIfAbsent(d, k ->
+                  IntStream.rangeClosed(1, d).boxed().collect(Collectors.toSet()));
```

Comment:
Here, we're writing to the map in a multi-threaded context. We should either use a concurrent hash map or change this logic such that it's read-only.

Reply:
@siddhantsangwan Thanks for suggestion ! It should fixed to concurrent hash map. 

Btw, I performed benchmarking for several methods (as following). I think using `map` is unnecessary here and alternative approaches  instead.